### DISCUSSION TITLE
Case insensitive query

### DISCUSSION THREAD

**Discussion by fschon** [opencrx/opencrx#77] [Q&A]
Is it possible to perform a case insensitive query. In my case I need to do a search for a realm with a all lowercase string. So for example I want to search for the realm 'Standard' with the text 'standard'. Some realms are CamelCased and so I cannot just ignore the first character.

```
RealmQuery realmQuery = (RealmQuery) pm.newQuery(Realm.class);
realmQuery.name().equalTo("standard");  <-- this needs to be a case insensitive search!
```

Thx for the help.

**Comment by brunoamacos** [comment]
Hi fschon

realmQuery.name().like("(?i)standard");

**Comment by fschon** [comment]
Thanks Bruno, works great. Do you know of any documentation for other this and other tricks?

**Comment by wfro64** [comment]
Please have a look at the classes in the following packages. There you will find many examples for all kind of queries:

* https://github.com/opencrx/opencrx/tree/master/core/src/main/java/org/opencrx/kernel/backend
* https://github.com/opencrx/opencrx/tree/master/core/src/data/org.opencrx/WEB-INF/classes/org/opencrx/portal/wizard
* https://github.com/opencrx/opencrx/tree/master/core/src/test/java/org/opencrx/kernel/api
